queue.go

```diff
@@ -0,0 +1,52 @@
+package cleisthenes
+
+// Define transaction FIFO queue
+type queue struct {
+	txs []Transaction
+}
+
+// EmptyQueErr is for calling peek(), poll() when queue is empty.
+type emptyQueErr struct {}
+
+func (e *emptyQueErr) Error() string {
+	return "queue is empty."
+}
+
+func (q *queue) newQueue() queue {
+	return queue {
+		txs: []Transaction{},
+	}
+}
+
+// Check whether queue is empty or not.
+func (q *queue) isEmpty() bool {
+	return len(q.txs) == 0
+}
+
+// Return first element of queue, but not erase it.
+func (q *queue) peek() (*Transaction, error) {
+	if q.isEmpty() {
+		return nil, &emptyQueErr{}
+	}
+	return &q.txs[0], nil
+}
+
+// Return first element of queue, and erase it.
+func (q *queue) poll() (*Transaction, error) {
+	ret, err := q.peek()
+	if err != nil {
+		return nil, err
+	}
+	q.txs = q.txs[1:]
```

Comment:
Is it safe that returning a pointer and erasing an original data of the pointer? I wonder about it!

Reply:
but If erasing original pointer, maybe we cannot use it.
so I am going to implement not using pointer.